src/plugins/explore/public/application/utils/state_management/utils/get_prompt_mode_is_available/get_prompt_mode_is_available.ts

```diff
@@ -16,19 +18,14 @@ export const getPromptModeIsAvailable = async (
       .getQueryEditorExtensionMap();
 
     // Check if query assist is available through data plugin extension system
-    if (!extensions['query-assist']) {
+    const queryAssistExtension = extensions['query-assist'];
+    if (!queryAssistExtension) {
       return false;
     }
 
-    // check if agent is available
-    const response: { configuredLanguages: string[] } = await services.http.get(
-      '/api/enhancements/assist/languages',
-      {
-        query: { dataSourceId: dataset.dataSource?.id },
-      }
+    return await firstValueFrom(
+      queryAssistExtension.isEnabled$({} as QueryEditorExtensionDependencies)
```

Comment:
not for this PR but we can probably get ride of the `dependencies` argument in the type definition later, since implementations shouldn't need it